compiler/modules/types/list.bal

```diff
@@ -582,58 +582,44 @@ function listSubtypeDiff(Context cx, SubtypeData t1, SubtypeData t2) returns Sub
     return b;
 }
 
-function isDisjoint(Context cx, Bdd t1, Bdd t2) returns boolean {
-    ListAtomicType[]? posAtoms = bddListAtoms(cx, t1);
-    ListAtomicType[]? negAtoms = bddListAtoms(cx, t2);
-    if posAtoms is () || negAtoms is () {
-        return false;
+function isTupleDisjointSimple(Context cx, Bdd t1, Bdd t2) returns boolean? {
+    if t1 !is BddNode || t1.left !is true || t1.middle !is false || t1.right !is false {
```

Comment:
Don't need to be this restrictive.

Just intersect the positive atoms, and if we get empty, return true.

Reply:
fixed